File: src/main/java/rr/language/inspections/NonExistingCulture.java
package rr.language.inspections;

import com.intellij.codeInspection.LocalInspectionToolSession;
import com.intellij.codeInspection.ProblemHighlightType;
import com.intellij.codeInspection.ProblemsHolder;
import org.jetbrains.annotations.NotNull;
import rr.language.RRUtil;
import rr.language.Util;
import rr.language.psi.*;

import java.util.Collection;

public class NonExistingCulture extends Inspector {
    @NotNull
    protected RRVisitor buildRRVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
        return new RRVisitor() {
            @Override
            public void visitCultureRef(@NotNull RRCultureRef element) {
                Collection<String> all = RRUtil.findAllCulturesAsStrings(element.getProject());

                if (!all.contains(element.getText())) {
                    holder.registerProblem(element, "Non existing culture", ProblemHighlightType.ERROR);
                }
            }

            public void visitStrCultureRef(@NotNull RRStrCultureRef element) {
                Collection<String> all = RRUtil.findAllCulturesAsStrings(element.getProject());

                if (!all.contains(Util.unquote(element.getText()))) {
                    holder.registerProblem(element, "Non existing culture", ProblemHighlightType.ERROR);
                }
            }
        };
    }
}

File: src/main/java/rr/language/inspections/NonExistingCulture.kt
package rr.language.inspections

import com.intellij.codeInspection.LocalInspectionToolSession
import com.intellij.codeInspection.ProblemHighlightType
import com.intellij.codeInspection.ProblemsHolder
import rr.language.RRUtil.findAllCulturesAsStrings
import rr.language.Util.unquote
import rr.language.psi.RRCultureRef
import rr.language.psi.RRStrCultureRef
import rr.language.psi.RRVisitor

class NonExistingCulture : Inspector() {
    override fun buildRRVisitor(holder: ProblemsHolder, session: LocalInspectionToolSession): RRVisitor {
        return object : RRVisitor() {
            override fun visitCultureRef(element: RRCultureRef) {
                val all = findAllCulturesAsStrings(element.project)
                if (!all.contains(element.text)) {
                    holder.registerProblem(element, "Non existing culture", ProblemHighlightType.ERROR)
                }
            }

            override fun visitStrCultureRef(element: RRStrCultureRef) {
                val all = findAllCulturesAsStrings(element.project)
                if (!all.contains(unquote(element.text))) {
                    holder.registerProblem(element, "Non existing culture", ProblemHighlightType.ERROR)
                }
            }
        }
    }
}